src/Integration/Suppression/SuppressedIssuesProvider.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core.SystemAbstractions;
+using SonarLint.VisualStudio.Integration.NewConnectedMode;
+using SonarQube.Client;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    [Export(typeof(ISonarQubeIssuesProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class SuppressedIssuesProvider : ISonarQubeIssuesProvider
+    {
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly Func<BindingConfiguration, ISonarQubeIssuesProvider> createProviderFunc;
+
+        private ISonarQubeIssuesProvider instance;
+        private bool disposed;
+
+        [ImportingConstructor]
+        public SuppressedIssuesProvider(IActiveSolutionBoundTracker activeSolutionBoundTracker,
+            ISonarQubeService sonarQubeService,
+            ILogger logger)
+            : this(activeSolutionBoundTracker,
```

Comment:
@duncanp-sonar what do you think? 🙈

Reply:
As discussed offline, it will be slightly more readable with a delegate instead of a function.
Other than that, let's go with this for the time being so we can move on.